pkg/apis/apm/v1/apmserver_types.go

```diff
@@ -13,7 +13,12 @@ import (
 	commonv1 "github.com/elastic/cloud-on-k8s/pkg/apis/common/v1"
 )
 
-const ApmServerContainerName = "apm-server"
+const (
+	ApmServerContainerName = "apm-server"
+	// Kind is inferred from the struct name using reflection in SchemeBuilder.Register()
+	// we duplicate it as a constant here for practical purposes.
+	Kind = "ApmServer"
```

Comment:
I did not find a better alternative to that. Using reflection similar to what SchemeBuilder.Register() is a bit more accurate but not sure if better? (we know those Kinds cannot change anyway? otherwise we'll run into all sorts of troubles)